components/motor/gpio/motor_encoder.go

```diff
@@ -354,184 +240,162 @@ func (m *EncodedMotor) rpmMonitor() {
+
+		lastPos = pos
 		lastTime = now
 	}
 }
 
-// return is if we are in a ramp phase.
-func (m *EncodedMotor) rpmMonitorPass(pos, lastPos, now, lastTime int64, rpmDebug bool) bool {
+// makeAdjustments does the math required to see if the RPM is too high or too low,
+// and if the goal position has been reached.
+func (m *EncodedMotor) makeAdjustments(pos, lastPos float64, now, lastTime int64) {
 	m.stateMu.Lock()
 	defer m.stateMu.Unlock()
 
-	var ticksLeft int64
-
-	currentRPM := m.computeRPM(pos, lastPos, now, lastTime)
-	m.state.currentRPM = currentRPM
-
-	if !m.state.regulated && math.Abs(m.state.desiredRPM) > 0.001 {
-		m.rpmMonitorPassSetRpmInLock(currentRPM, m.state.desiredRPM, -1, rpmDebug)
-		return false
+	// calculate RPM based on change in position and change in time
+	deltaPos := (pos - lastPos) / float64(m.ticksPerRotation)
+	deltaTime := (float64(now) - float64(lastTime)) / float64(6e10)
+	if deltaTime == 0.0 {
+		m.state.currentRPM = 0
+	} else {
+		m.state.currentRPM = deltaPos / deltaTime
 	}
 
-	if !m.state.regulated {
-		return false
-	}
-
-	// correctly set the ticksLeft accounting for power supplied to the motor and the expected direction of the motor
-	ticksLeft = (m.state.setPoint - pos) * sign(m.state.lastPowerPct) * m.flip
-	rotationsLeft := float64(ticksLeft) / float64(m.ticksPerRotation)
+	dir := m.directionMovingInLock()
 
-	if rotationsLeft <= 0 { // if we have reached goal or overshot, turn off
-		if rpmDebug {
-			m.logger.Debugf("rot %.2f, stopping motor", rotationsLeft)
+	// "controls"
+	quartRot := float64(m.ticksPerRotation) / 4.0
+	if pos < m.state.goalPos+quartRot && pos > m.state.goalPos-quartRot {
+		powerPct := m.state.lastPowerPct - (m.rampRate * 2 * m.directionMovingInLock())
+		if sign(powerPct) != m.directionMovingInLock() {
+			powerPct *= -1
 		}
-		err := m.off(m.cancelCtx)
-		if err != nil {
-			m.logger.Warnf("error turning motor off from after hit set point: %v", err)
+		m.state.lastPowerPct = powerPct
+		if err := m.setPower(m.cancelCtx, powerPct, true); err != nil {
+			panic(err)
 		}
-		return false
-	}
-
-	// slow down so we don't overshoot
-	// halve and quarter rpm values based on seconds remaining in move
-
-	desiredRPM := m.state.desiredRPM
-	timeLeftSeconds := math.Abs(60.0 * rotationsLeft / desiredRPM)
-
-	desiredRPM = slowDownMath(timeLeftSeconds, desiredRPM, m.rampRate)
-
-	if rpmDebug {
-		m.logger.Debugf(" - rotationsLeft %.2f timeLeftSeconds %.2f rpm(%v -> %v)",
-			rotationsLeft, timeLeftSeconds, m.state.desiredRPM, desiredRPM)
-	}
-
-	m.rpmMonitorPassSetRpmInLock(currentRPM, desiredRPM, rotationsLeft, rpmDebug)
-
-	return true
-}
-
-// TODO(erh): someone make this better.
-func slowDownMath(timeLeftSeconds, desiredRPM, rampRate float64) float64 {
-	if timeLeftSeconds <= 0 {
-		return desiredRPM
-	}
-
-	if timeLeftSeconds < .5 {
-		desiredRPM *= math.Min(1, rampRate/.5)
+		return
 	}
+	m.logger.Debugf("currentRPM = %v, desiredRPM = %v", m.state.currentRPM, m.state.desiredRPM)
```

Comment:
This will spam a lot when we have debug on - so we used to hide it under the rpmDebug flag, might be worth it to do so with the config attribute here then.

Reply:
I'll remove it for now and revisit when I add rpmDebug back